sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -59,12 +64,43 @@
   final Context context;
   private final SentryOptions options;
 
+  private final Future<Map<String, Object>> contextData;
+
   public DefaultAndroidEventProcessor(Context context, SentryOptions options) {
     this.context =
         Objects.requireNonNull(
             context != null ? context.getApplicationContext() : null,
             "The application context is required.");
     this.options = Objects.requireNonNull(options, "The SentryOptions is required.");
+
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    contextData = executorService.submit(() -> loadContextData());
+
+    executorService.shutdown();
+  }
+
+  private Map<String, Object> loadContextData() {
+    Map<String, Object> map = new HashMap<>();
+    String[] proGuardUuids = getProGuardUuids();
+    if (proGuardUuids != null) {
+      map.put("proGuardUuids", proGuardUuids);
+    }
+
+    map.put("rooted", isRooted());
+
+    String androidId = getAndroidId();
+    if (androidId != null) {
+      map.put("androidId", androidId);
```

Comment:
What about having the map keys as constants? To get compiler checks here and make sure they don't go out of sync?